CONTRIBUTING.md

```diff
@@ -226,83 +208,78 @@ Ensure that you base and target your PR on the `master` branch.
 All feature additions should be targeted against `master`. Bug fixes for an outstanding release candidate
 should be targeted against the release candidate branch.
 
+If needed, we backport a commit from `master` to the release branch (not consensus breaking feature, optimization...)..
+
 ### Development Procedure
 
 - the latest state of development is on `master`
 - `master` must never fail `make lint test test-race`
-- `master` should not fail `make lint`
 - no `--force` onto `master` (except when reverting a broken commit, which should seldom happen)
 - create a development branch either on github.com/cosmos/cosmos-sdk, or your fork (using `git remote add origin`)
-- before submitting a pull request, begin `git rebase` on top of `master`
 
-### Pull Merge Procedure
+Submitting a pull request:
+- before submitting a pull request, merge the latest master `git merge origin/master`,
+- run `make lint test` to ensure that all checks and tests pass,
+- GitHub will squash commits and rebase on top of the master.
+
+Finalizing pull requests
+- ask contributors for review,
+- respond to all comments,
+- resolve merge conflicts
+
+Pull requests are merged automatically using `automerge` action.
 
-- ensure pull branch is rebased on `master`
-- run `make test` to ensure that all tests pass
-- merge pull request
+### Major Release Procedure
 
-### Release Procedure
+A _major release_ is an increment of the first number (eg: `v1.2` → `v2.0.0`) or the point number (eg: `v1.1 → v1.2.0`).
+
+TL;DR before making a new _major_ release we do beta and release candidate releases. For example, for release 1.0.0:
+```
+v1.0.0-beta1 → v1.0.0-beta2 → ... → v1.0.0-rc1 → v1.0.0-rc2 → ... → v1.0.0
+```
 
-- Start on `master`
-- Create the release candidate branch `rc/v*` (going forward known as **RC**)
-  and ensure it's protected against pushing from anyone except the release
-  manager/coordinator
+- Release a first beta version on `master` branch and freeze `master` for receiving any new features. Once beta is released, we focus on releasing the release candidate:
+  - finish audits and reviews
+  - kick off a large round of simulation testing (e.g. 400 seeds for 2k blocks)
+  - perform functional tests
+  - add more tests
+  - release new beta version as the bugs are discovered and fixed.
+- Once the team feels that the `master` works fine we create a `release/vY` branch (going forward known a release branch), where `Y` is the version number, with the patch part substituted to `x` (eg: 0.42.x, 1.0.x). Ensure the release branch is protected against pushing from anyone except the release manager/coordinator.
```

Comment:
```suggestion
- After the team feels that the `master` works fine we create a `release/vY` branch (going forward known a release branch), where `Y` is the version number, with the patch part substituted to `x` (eg: 0.42.x, 1.0.x). Ensure the release branch is protected so that pushes against the release branch are permitted only by the release manager or release coordinator.
```